kubetest/aksengine.go

```diff
@@ -1101,7 +1102,7 @@ func (c *aksEngineDeployer) Build(b buildStrategy) error {
 		if c.customKubeBinaryURL, err = c.uploadToAzureStorage(newK8sNodeTarball); err != nil {
 			return err
 		}
-	} else if !*testCcm && !*testAzureDiskCSIDriver && !*testAzureFileCSIDriver && !*testBlobCSIDriver && !*testSecretStoreCSIDriver && !*testSMBCSIDriver && !*testNFSCSIDriver && !strings.EqualFold(string(b), "none") {
+	} else if (!*testCcm && !*testAzureDiskCSIDriver && !*testAzureFileCSIDriver && !*testBlobCSIDriver && !*testSecretStoreCSIDriver && !*testSMBCSIDriver && !*testNFSCSIDriver && !strings.EqualFold(string(b), "none")) || *runExternalE2EGinkgoTest {
```

Comment:
`--storage-testdriver-repo-path` is always with a test driver, for this PR, it's `--test-azure-disk-csi-driver`, so it's not necessary here?

Reply:
The above check only builds the required components (e2e.test, kubectl, ginkgo) when none of the following args <test-ccm, test-azure-file-csi-driver, test-azure-disk-csi-driver, test-blob-csi-driver, test-secrets-store-csi-driver, test-csi-driver-smb, test-csi-driver-nfs> are present. (since its checking for false)

Since, we want to build the e2e.test before running the external e2e k8s test, shouldn't we need to build it before running?